src/app/auth-service.ts

```diff
@@ -1,9 +1,53 @@
 import { Injectable } from '@angular/core';
+import "firebase/auth";
+import { createUserWithEmailAndPassword, getAuth, signInWithEmailAndPassword, onAuthStateChanged } from "firebase/auth"
+import {  Router } from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-    token:string = '12345';
-  constructor() { }
+  isLoggedIn: boolean = false;
+  constructor(private router: Router) { }
+
+  registerUser(email: string, password: string) {
+    const auth = getAuth();
+    createUserWithEmailAndPassword(auth, email, password)
+    .then( res => {
+      console.log(res)
```

Comment:
Invoking promise should be done on component not in service >> We will show an error from service ???

Reply:
Yes, we are throwing an error when register/login fails.